processing/src/test/java/io/druid/query/search/SearchQueryRunnerTest.java

```diff
@@ -342,34 +342,34 @@ public void testSearchWithExtractionFilter1()
         true,
         null,
         true,
-        false
+        true
     );
 
-    checkSearchQuery(
-        Druids.newSearchQueryBuilder()
-              .dataSource(QueryRunnerTestHelper.dataSource)
-              .granularity(QueryRunnerTestHelper.allGran)
-              .filters(
-                  new ExtractionDimFilter(
-                      QueryRunnerTestHelper.qualityDimension,
-                      automotiveSnowman,
-                      lookupExtractionFn,
-                      null
-                  )
-              )
-              .intervals(QueryRunnerTestHelper.fullOnInterval)
-              .dimensions(
-                  new ExtractionDimensionSpec(
-                      QueryRunnerTestHelper.qualityDimension,
-                      null,
-                      lookupExtractionFn,
-                      null
-                  )
-              )
-              .query("☃")
-              .build(),
-        expectedHits
-    );
+    SearchQuery query = Druids.newSearchQueryBuilder()
+                              .dataSource(QueryRunnerTestHelper.dataSource)
+                              .granularity(QueryRunnerTestHelper.allGran)
+                              .filters(
+                                  new ExtractionDimFilter(
+                                      QueryRunnerTestHelper.qualityDimension,
+                                      automotiveSnowman,
+                                      lookupExtractionFn,
+                                      null
+                                  )
+                              )
+                              .intervals(QueryRunnerTestHelper.fullOnInterval)
+                              .dimensions(
+                                  new ExtractionDimensionSpec(
+                                      QueryRunnerTestHelper.qualityDimension,
+                                      null,
+                                      lookupExtractionFn,
+                                      null
+                                  )
+                              )
+                              .query("☃")
+                              .build();
+
+    checkSearchQuery(query, expectedHits);
+    checkSearchQuery(query, toolChest.mergeResults(toolChest.preMergeQueryDecoration(runner)), expectedHits);
```

Comment:
If you add this style of checking to the base `checkSearchQuery` (so it always does both checks – one on a bare runner and one on the decorated runner) then that'll improve the coverage of all search tests, which would be cool.

might as well do a postMergeQueryDecoration too.